### DISCUSSION TITLE
pre-commit and github action generators!

### DISCUSSION THREAD

**Discussion by jdx** [jdx/mise#2160] [Announcements]
I've added a [couple of simple commands](https://mise.jdx.dev/cli/#mise-generate-git-pre-commit-options) to mise I would love some feedback on, `mise generate git-pre-commit` and `mise generate github-action`.

While far from complex, I do think these sort of things are great for quickly bootstrapping a project and setting up basic good practices. I'd love any and all feedback on them, give them a try!

**Comment by mustafa0x** [comment]
Brilliant. What args are passed to the task? lefthook passes the following, https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#run

**Reply by jdx** [reply]
I don't think arguments would be a good way to pass information. Nothing is passed now but we could add similar things to env vars.

**Reply by jdx** [reply]
I think this would be fairly easy to do, we'd just need to change the script it outputs to something like this:

```bash
#!/usr/bin/env bash
CHANGED_FILES="$(git ls-files ...)" # idk what the right args off the top of my head are
export CHANGED_FILES

mise run pre-commit
```

**Reply by mustafa0x** [reply]
`git diff-tree -r --name-only --no-commit-id ORIG_HEAD HEAD`, I recently learned
